src/containers/audio-rewards-page/WalletModal.tsx

```diff
@@ -85,56 +92,67 @@ const AddWalletTitle = () => {
 
 const titlesMap = {
   CONNECT_WALLETS: {
-    ADD_WALLET: <AddWalletTitle />,
-    REMOVE_WALLET: messages.removeWallets,
-    ERROR: messages.sendError
+    ADD_WALLET: () => <AddWalletTitle />,
+    REMOVE_WALLET: () => messages.removeWallets,
+    ERROR: () => messages.sendError
   },
   RECEIVE: {
-    KEY_DISPLAY: (
-      <TitleWrapper label={messages.receive}>
-        <IconReceive className={styles.receiveWrapper} />
-      </TitleWrapper>
-    )
+    KEY_DISPLAY: () => {
+      const useSolSPLAudio = getRemoteVar(BooleanKeys.USE_SPL_AUDIO) as boolean
+      return (
+        <TitleWrapper
+          label={useSolSPLAudio ? messages.receiveSPL : messages.receive}
+        >
+          <IconReceive className={styles.receiveWrapper} />
+        </TitleWrapper>
+      )
+    }
```

Comment:
nit: would this work as an IIFE, and we could avoid having to wrap everything in a function?

Reply:
the getRemoteVar would be invoked immediately and will not have the right value - which was my first attempt. 
I agree though that it's ugly.

Reply:
we have a wait for remote config helper https://github.dev/AudiusProject/audius-client/blob/637c9b4f337c117ee54ecead959693e77990e825/src/services/remote-config/Provider.ts#L173

useful?

Reply:
it's async, so don't think there's a clean to use that here

Reply:
it's async, so don't think there's a clean to use that here